### DISCUSSION TITLE
Trying to build for 7170

### DISCUSSION THREAD

**Discussion by jschwartzenberg** [Freetz-NG/freetz-ng#913] [Allgemein] [answered]
Should the older models still work?

I'm running into this:
```
/home/julius/checkouts/freetz-ng/toolchain/build/mipsel_gcc-5.5.0_uClibc-0.9.29/mipsel-linux-uclibc/bin/mipsel-linux-uclibc-ld --no-as-needed -EL -melf32ltsmip -shared --warn-common --warn-once -z combreloc -z relro -z defs  -init __uClibc_init -soname=libc.so.0 -o lib/libuClibc-0.9.29.so  --whole-archive libc/libc_so.a --no-whole-archive ./lib/interp.os ./lib/ld-uClibc.so.0 ./lib/uclibc_nonshared.a /home/julius/checkouts/freetz-ng/toolchain/build/mipsel_gcc-5.5.0_uClibc-0.9.29/mipsel-linux-uclibc/bin/../lib/gcc/mipsel-linux-uclibc/5.5.0/libgcc.a 
libc/libc_so.a(libc_pthread_init.oS): In function `testandset':
libc_pthread_init.c:(.text+0x0): multiple definition of `testandset'
libc/libc_so.a(forward.oS):forward.c:(.text+0x0): first defined here
libc/libc_so.a(libc_pthread_init.oS): In function `__compare_and_swap':
libc_pthread_init.c:(.text+0x24): multiple definition of `__compare_and_swap'
libc/libc_so.a(forward.oS):forward.c:(.text+0x24): first defined here
libc/libc_so.a(rpc_thread.oS): In function `testandset':
rpc_thread.c:(.text+0x34): multiple definition of `testandset'
libc/libc_so.a(forward.oS):forward.c:(.text+0x0): first defined here
libc/libc_so.a(rpc_thread.oS): In function `__compare_and_swap':
rpc_thread.c:(.text+0x58): multiple definition of `__compare_and_swap'
libc/libc_so.a(forward.oS):forward.c:(.text+0x24): first defined here
make[3]: *** [libc/Makefile.in:59: lib/libc.so] Error 1
make[2]: *** [Makerules:203: lib/libc.so.0] Fout 2
make[1]: *** [make/toolchain/target/uclibc/uclibc.mk:180: /home/julius/checkouts/freetz-ng/source/toolchain-mipsel_gcc-5.5.0_uClibc-0.9.29/uClibc-0.9.29/lib/libc.a] Fout 2
make: *** [Makefile:47: envira] Beëindigd
```

Could anybody give a hint on how to handle this? I understand issues cannot be addressed anymore? Thanks a lot in advance!!

**Comment by fda77** [comment, marked as answer]
yes, works great with default gcc v4. I've disabled the gcc v5 you used as im to lazy to do any investigations
914d022f123ab47638f8d34f7bb34af3f355a871

**Reply by jschwartzenberg** [reply]
Yep indeed working great now! Many thanks! You're the best!!

**Comment by harryboo** [comment]
@fda77 : Please revert the 3 commits ( 914d022f123ab47638f8d34f7bb34af3f355a871, d99541a4fc8289f5c2ea6f31b21d935ffee64650, deb5c9f0d78b77480b33e86f0324dc150dbe7364 ) as this is a configuration option and everyone can choose (test) the right one
or
implement this [fix](https://lore.kernel.org/buildroot/485A5E5B.1020907@cetrtapot.si/)

**Reply by fda77** [reply]
Freetz replaces older uclibc version of avm, so its always risky to patch it. And i dont want test, even i have no device to d so etc...
And i never understand why people want to use another eg gcc version!
97e3a386f 
I hope this helps you and prevents switching from a newer device with gcc5 to an older and the unsupported gcc version keeps selected

**Reply by jschwartzenberg** [reply]
@harryboo if you create a branch with this implemented, I'm willing to test your fix. It will not be soon though as I won't have access to my 7170 for the upcoming weeks.

I understand @fda77's concern though, so it would be good to know if there are specific advantages to this.
